src/app/common/directives/header-anchor.directive.ts

```diff
@@ -0,0 +1,36 @@
+import {AfterViewInit, Directive, ElementRef, Input, Renderer2} from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+
+@Directive({
+  selector: 'h3,h4',
+})
+export class HeaderAnchorDirective implements AfterViewInit {
+
+  @Input('id') public overriddenId: string | undefined;
```

Comment:
I would just call it `id` since it is only overridden if no id was explicitly defined.